go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -174,17 +222,65 @@ type ExternalRouteInfo struct {
 	PodExternalRoutes map[string]map[string]string
 }
 
-// This structure contains the processed information of a policy.
-// This information is then used to update the network components (North Bound DB, conntrack) by applying the IPs here to each of the target namespaces defined in the from field.
-type routePolicy struct {
-	// targetNamespacesSelector contains the namespace selector defined in the from field in the policy.
-	targetNamespacesSelector *metav1.LabelSelector
+// routePolicyState contains current policy state as it was applied.
+// Since every config is applied to a pod, podInfo stores current state for every target pod.
+type routePolicyState struct {
+	// namespaceName: podName: configured gateways for the pod
+	targetNamespaces map[string]map[ktypes.NamespacedName]*podInfo
+}
+
+func newRoutePolicyState() *routePolicyState {
+	return &routePolicyState{
+		targetNamespaces: map[string]map[ktypes.NamespacedName]*podInfo{},
+	}
+}
+
+// Equal compares StaticGateways and DynamicGateways elements for every namespace to be exactly the same,
+// including failedAdd status
+func (rp *routePolicyState) Equal(rp2 *routePolicyState) bool {
+	for nsName, nsInfo := range rp.targetNamespaces {
+		nsInfo2, found := rp2.targetNamespaces[nsName]
+		if !found {
+			return false
+		}
+		for podName, podInfo := range nsInfo {
+			podInfo2, found := nsInfo2[podName]
+			if !found {
+				return false
+			}
+			if !podInfo.StaticGateways.Equal(podInfo2.StaticGateways) ||
```

Comment:
suggestion: replace with
```
if !reflect.DeepEqual(podInfo, podInfo2) {
```

Reply:
I don't think we can do that, because `gatewayInfoList` is ordered, but we want to compare it as set

Reply:
Does the order matter? I didn't think it would but if it does then let's keep it this way.

Reply:
it's the opposite: deepEqual checks the ordering, but we don't want that, so `(g *gatewayInfoList) Equal` method ignores ordering